### DISCUSSION TITLE
Switching to Microsoft.Extensions.Logging

### DISCUSSION THREAD

**Discussion by Cheesebaron** [MvvmCross/MvvmCross#4087] [Ideas]
We have our own IMvxLog and log providers for Serilog, Log4Net etc. which depends on a bunch of reflection to figure out whether any of these logging libraries are added in your project.

I  dread the day that one of these libraries change enough to someone has to go in and fix the log providers, because the code is not, lets say that clear and easily maintainable.

I've been using ILogger and ILoggerFactory from Microsoft.Extensions.Logging in a bunch of projects now, and it is so easy to interface it with Serilog etc., since there are packages for each of these that integrate nicely.

The thought here is to take a hard dependency on Microsoft.Extensions.Logging and entirely ditch IMvxLog and IMvxLogProvider.

Pros:
- Easy integration with Serilog, Log4Net etc.
- Less maintenance
- No reflection

Cons:
- Breaking change

Please state your concerns or support in comments.

**Comment by MartinZikmund** [comment]
I fully support this idea!
